client/my-sites/checkout/src/components/domain-registration-hsts.tsx

```diff
@@ -1,40 +1,45 @@
 import { localizeUrl } from '@automattic/i18n-utils';
-import { localize } from 'i18n-calypso';
-import { isEmpty, merge, reduce } from 'lodash';
+import { LocalizeProps, localize } from 'i18n-calypso';
 import { PureComponent } from 'react';
 import { connect } from 'react-redux';
 import { getDomainRegistrations, getDomainTransfers } from 'calypso/lib/cart-values/cart-items';
 import { getTld, isHstsRequired } from 'calypso/lib/domains';
 import { HTTPS_SSL } from 'calypso/lib/url/support';
 import CheckoutTermsItem from 'calypso/my-sites/checkout/src/components/checkout-terms-item';
 import { getProductsList } from 'calypso/state/products-list/selectors';
+import type { ResponseCart } from '@automattic/shopping-cart';
+import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
 
-/* eslint-disable wpcalypso/jsx-classname-namespace */
+export interface DomainRegistrationHstsProps {
+	cart: ResponseCart;
+}
+
+export interface DomainRegistrationConnectedHstsProps {
+	productsList: Record< string, ProductListItem >;
+}
```

Comment:
Those interfaces are not used outside of this file, what's the reason for exporting them?

Reply:
I find it's helpful to always export interfaces for public functions and components so that it's possible to have access to their types if they are needed outside the file. For example, if someone wanted to make a wrapper for this component and needed to know its props, exporting `DomainRegistrationHstsProps` would allow that; otherwise you'd have to jump through hoops to dynamically reference the props of the component. This issue has caused me a lot of work in the past.

You're right that I don't need to export `DomainRegistrationConnectedHstsProps`. That was just habit.

Do you know of a downside to exporting types like this?

Reply:
I was just curious about the general approach in Calypso for it.

Exporting the component props by default seems fine, thanks!